### DISCUSSION TITLE
moment.duration(x).toISOString() not returning in this format 'PnYnMnDTnHnMnS'

### DISCUSSION THREAD

**Discussion by victorzw895** [moment/moment#6423] [Q&A] [answered]
docs say that moment.duration(x).toISOString()
will

```
Returns duration in string as specified by [ISO 8601 standard](https://en.wikipedia.org/wiki/ISO_8601#Durations).

moment.duration(1, 'd').toISOString() // "P1D"
Format PnYnMnDTnHnMnS description:
```

However i am only getting the hours 'PT15528H' for 55900800000 milliseconds when it should be something like 'P1Y9M7D'

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the report! `PT15528H` is valid ISO 8601. A raw millisecond count cannot be unambiguously converted into calendar years or months because those units have variable lengths. If those components matter, I'd construct the duration from explicit calendar units or endpoints. I'm closing this as answered; the duration cannot safely invent a calendar origin.
